app/src/main/java/com/example/tasky/feature_agenda/presentation/agenda_screen/AgendaViewModel.kt

```diff
@@ -110,27 +111,31 @@ class AgendaViewModel @Inject constructor(
                 }
             }
 
-            is AgendaEvent.ToggleItemCreationMenu -> {
+            AgendaEvent.ToggleItemCreationMenu -> {
                 _state.update {
                     it.copy(isItemCreationMenuVisible = !state.value.isItemCreationMenuVisible)
                 }
+                Log.d("MENU", "The updated state is ${state.value}")
             }
 
             is AgendaEvent.ToggleIndividualItemMenu -> {
+                Log.d("MENU", "ToggleItemVisibility triggered in viewModel")
                 menuVisibilityOptions[event.itemKey] = event.showIndividualMenu
+                Log.d("MENU", "The updated map is: $menuVisibilityOptions")
                 _state.update {
                     it.copy(
                         isItemMenuVisible = menuVisibilityOptions
                     )
                 }
```

Comment:
As long as the object reference is the same to the map, changes won't recompose the UI. You'd either need to make a complete copy of the map (by rebuilding it) or you use a `mutableStateMapOf`. That's also why you are not seeing this take effect on the UI